src/testcontainers/src/generic-container/generic-container.test.ts

```diff
@@ -434,4 +435,39 @@ describe("GenericContainer", () => {
     await expect(stopContainerPromises).resolves.not.toThrow();
     expect(await getRunningContainerNames()).not.toContain(container.getName());
   });
+
+  it("should build a target stage", async () => {
+    const context = path.resolve(fixtures, "docker-multi-stage");
+    const firstConatiner = await GenericContainer.fromDockerfile(context).withTarget("first").build();
+    const secondContainer = await GenericContainer.fromDockerfile(context).withTarget("second").build();
+
+    const firstStartedContainer = await firstConatiner.start();
+    const secondStartedContainer = await secondContainer.start();
+
+    const logLines: Array<string> = [];
+
+    const recordLogs = async (name: string, container: StartedTestContainer) => {
```

Comment:
Is there an easier way to tell the difference between both containers, e.g could each set a label?